ZFSin/zfs/module/zfs/vdev_disk.c

```diff
@@ -439,6 +419,85 @@ vdev_disk_ioctl_done(void *zio_arg, int error)
 	zio_interrupt(zio);
 }
 
+struct vdev_disk_callback_struct {
+	zio_t *zio;
+	PIRP irp;
+	void *b_addr;
+	char work_item[0];
+};
+typedef struct vdev_disk_callback_struct vd_callback_t;
+
+static void
+vdev_disk_io_start_done(void *param)
+{
+	vd_callback_t *vb = (vd_callback_t *)param;
+
+	ASSERT(vb != NULL);
+
+	NTSTATUS status = vb->irp->IoStatus.Status;
+	zio_t *zio = vb->zio;
+	zio->io_error = (!NT_SUCCESS(status) ? EIO : 0);
+
+	// Return abd buf
+	if (zio->io_type == ZIO_TYPE_READ) {
+		VERIFY3S(zio->io_abd->abd_size, >= , zio->io_size);
+		abd_return_buf_copy_off(zio->io_abd, vb->b_addr,
+			0, zio->io_size, zio->io_abd->abd_size);
+	} else {
+		VERIFY3S(zio->io_abd->abd_size, >= , zio->io_size);
+		abd_return_buf_off(zio->io_abd, vb->b_addr,
+			0, zio->io_size, zio->io_abd->abd_size);
+	}
+
+	UnlockAndFreeMdl(vb->irp->MdlAddress);
+	zio_delay_interrupt(zio);
```

Comment:
Can we make the `zio_delay_interrupt()` call be the last in the function.

Reply:
sure, will do.